src/Init/Data/Bool.lean

```diff
@@ -12,6 +12,42 @@ public section
 
 namespace Bool
 
+/--
+Boolean “logical and”. `land x y`.
+
+`land x y` is `true` when both of `x` or `y` are `true`. It is functionally the same as
+`x && y` but it does not have short-circuiting behavior: any call to `land` will evaluate both
+arguments. It uses non-branching code at runtime.
+
+Examples:
+ * `land false false = false`
+ * `land true false = false`
+ * `land false true = false`
+ * `land true true = true`
+-/
+@[expose, extern "lean_bool_land"]
+def land (x y : Bool) : Bool := x && y
```

Comment:
Is there a name that makes it easier to understand the difference between this and `Bool.and` at a glance? Maybe something like `strictAnd`?

Reply:
`strictAnd` already exists. It ensures that both arguments get evaluated but still has a branching implementation.

Reply:
I changed the definition of `strictAnd` and `strictOr` to be non-branching and I added `strictXor`.